frb_rust/src/rust2dart.rs

```diff
@@ -38,23 +37,25 @@ impl Rust2Dart {
     }
 
     /// Send an error back to the specified port.
-    pub fn error(&self, error_code: String, error_message: String) -> bool {
-        self.error_full(error_code, error_message, ())
+    pub fn error(&self, e: impl IntoDart) -> bool {
+        self.error_full(e)
+    }
+
+    /// Send a panic back to the specified port.
+    pub fn panic(&self, e: impl IntoDart) -> bool {
+        self.panic_full(e)
     }
 
     /// Send a detailed error back to the specified port.
-    pub fn error_full(
-        &self,
-        error_code: String,
-        error_message: String,
-        error_details: impl IntoDart,
-    ) -> bool {
-        self.isolate.post(vec![
-            RUST2DART_ACTION_ERROR.into_dart(),
-            error_code.into_dart(),
-            error_message.into_dart(),
-            error_details.into_dart(),
-        ])
+    pub fn error_full(&self, e: impl IntoDart) -> bool {
+        self.isolate
+            .post(vec![RUST2DART_ACTION_ERROR.into_dart(), e.into_dart()])
+    }
+
+    /// Send a detailed error back to the specified port.
+    pub fn panic_full(&self, e: impl IntoDart) -> bool {
+        self.isolate
+            .post(vec![RUST2DART_ACTION_PANIC.into_dart(), e.into_dart()])
```

Comment:
nit: In the old code, the function signature of `error` is different from `error_full`, so there are two functions. But now we have exactly the same. So maybe there is no need to have a `..._full` function at all